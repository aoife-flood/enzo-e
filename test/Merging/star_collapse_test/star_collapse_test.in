 Adapt {
     max_level = 0;
     min_level = 0;
 }

 Boundary {
     type = "periodic";
 }


Domain {
     lower = [ -6.172e24, -6.172e24, -6.172e24];
     rank = 3;
     upper = [ 6.172e24, 6.172e24, 6.172e24];
 }

 Field {
     alignment = 8;
     gamma = 1.0000001;
     ghost_depth = 4;
     courant = 0.3;
     history = 1;
     list = [ "density", "velocity_x", "velocity_y", "velocity_z", "acceleration_x", "acceleration_y", "acceleration_z", "total_energy", "internal_energy", "pressure", "density_total", "density_particle", "density_particle_accumulate", "potential", "density_gas", "X", "B", "X_copy", "B_copy", "potential_copy","diagonal" ];
#     list = [ "acceleration_x", "acceleration_y", "acceleration_z",
#     	      "density_particle", "density_total",
#	      "density_particle_accumulate", "potential", "potential_temp",
#	     "potential_copy", "X", "B", "X_copy", "B_copy" ,"diagonal"];
     padding = 0;
     dual_energy = true;
     diffusion = true;
 }

 Initial {
     collapse_stars {
     	 offset_factor = 0.1;
         infall_speed=6.3e7;
     	 centre = [1.543e24,1.543e24,1.543e24];
	 drift_velocity = [0.0,0.0,0.0];
	 truncation_radius = 3.086e24;
	 density = 1.84e-27;
     };
     list = [ "collapse_stars" ];
 }

 Mesh {
     root_blocks = [ 4, 4, 4 ];
     root_rank = 3;
     root_size = [64, 64, 64];
 }


Particle {
    list = ["star"];
    batch_size = 20000;
    star {
        attributes = [ "x", "double",
                       "y", "double",
                       "z", "double",
                       "vx", "double",
                       "vy", "double",
                       "vz", "double",
                       "ax", "double",
                       "ay", "double",
                       "az", "double",
                       "mass", "double",
		       "prevmass","double",
                       "creation_time", "double",
                       "lifetime", "double",
                       "metal_fraction", "double",
		       "accretion_rate", "double",
		       "accretion_rate_time", "double" ,
		       "class", "int32",
		       "timeindex", "int32",
		       "is_copy", "int64"];
        position = [ "x", "y", "z" ];
        velocity = [ "vx", "vy", "vz" ];
        group_list = "has_mass";
    }
}




 Method {
 #  check_gravity {
   #     particle_type = "star";
   # };
     gravity {
         solver = "bcg";
     };

     list = [ "pm_update", "merge_stars"];		
     pm_update {
         max_dt = 1e14;
     };

     merge_stars {

     merging_radius_cells = 0.3;
     
     };

 }

 Output {
   
     data {
         dir = [ "./Dir_Collapse-STARS_%04d", "cycle" ];
         field_list = [ "acceleration_x", "acceleration_y", "acceleration_z",
	 	      	"density_total", "potential" ];
	 particle_list = ["star"];
         name = [ "data-%04d-%04d.h5", "cycle", "proc" ];
         schedule {
             step = 1.0e15;
             var = "time";
         };
         type = "data";
     };
    
     list = ["data"];
     
 }

 Solver {
     bcg {
         iter_max = 200;
         monitor_iter = 1;
         precondition = "diagonal_D";
         res_tol = 0.01000000000000000;
         type = "bicgstab";
     };
     diagonal {
         type = "diagonal";
     };
     list = [ "bcg", "diagonal" ];
 }

 Stopping {
   	time = 4.89e16;
 }