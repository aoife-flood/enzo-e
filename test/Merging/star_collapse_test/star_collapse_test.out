execve failed to start process "/home/aoife/codes/enzo-e//bin/enzo-e": No such file or directory
